be/src/formats/orc/orc_chunk_reader.cpp

```diff
@@ -1161,7 +1161,7 @@ ColumnPtr OrcChunkReader::get_row_delete_filter(const std::set<int64_t>& deleted
     ColumnPtr filter_column = BooleanColumn::create(num_rows, 1);
     auto& filter = static_cast<BooleanColumn*>(filter_column.get())->get_data();
     auto iter = deleted_pos.lower_bound(start_pos);
-    auto end = deleted_pos.upper_bound(start_pos + num_rows);
+    auto end = deleted_pos.upper_bound(start_pos + num_rows - 1);
```

Comment:
Is it posible that the `start_pos` and `num_rows` are 0 ?

Reply:
before call `get_row_delete_filter()`, we will call `_orc_reader->read_next()` first. If num_rows is 0, `read_next()` function will return EOF directly, the code can't go here.

Reply:
> before call `get_row_delete_filter()`, we will call `_orc_reader->read_next()` first. If num_rows is 0, `read_next()` function will return EOF directly, the code can't go here.

OK